contrib/src/main/java/com/datatorrent/contrib/enrich/FixedWidthFSLoader.java

```diff
@@ -0,0 +1,248 @@
+  }
+
+  /**
+   * Sets fieldDescription i.e a String specifying information related to fields
+   * in fixed-width file. Format is [NAME]:[FIELD_TYPE]:[WIDTH]:[date format if
+   * FIELD_TYPE is DATE] FIELD_TYPE can take on of the values of
+   * {@link FIELD_TYPE}
+   * e.g.Year:INTEGER:4,Make:STRING:5,Model:STRING:40,Description:STRING:40,
+   * Price:DOUBLE:8,Date:DATE:10:\"dd:mm:yyyy\" Date format needs to be within
+   * quotes (" ")
+   * 
+   * @param fieldDescription
+   */
+  public void setFieldDescription(String fieldDescription)
+  {
+    this.fieldDescription = fieldDescription;
+  }
+
+  /**
+   * Gets the character used for padding in the fixed-width file.Default is
+   * white space (' ')
+   * 
+   * @return
+   */
+  public char getPadding()
+  {
+    return padding;
+  }
+
+  /**
+   * Sets the character used for padding in fixed-width file.Default is white
+   * space (' ')
+   * 
+   * @param padding
+   */
+  public void setPadding(char padding)
+  {
+    this.padding = padding;
+  }
+
+  /**
+   * Extracts the fields from a fixed width record and returns a map containing
+   * field names and values
+   */
+  @Override
+  Map<String, Object> extractFields(String line)
+  {
+    if (!initialized) {
+      init();
+      initialized = true;
+    }
+    String[] values = fixedWidthParser.parseLine(line);
+    if (hasHeader && Arrays.deepEquals(values, header)) {
+      return null;
+    }
+    Map<String, Object> map = Maps.newHashMap();
+    int i = 0;
+    for (Field field : fields) {
+      map.put(field.getName(), getValue(field, values[i++]));
+    }
+    return map;
+  }
+
+  private void init()
+  {
+    fields = new ArrayList<Field>();
+    List<String> headers = new ArrayList<String>();
+    List<Integer> fieldWidth = new ArrayList<Integer>();
+    for (String tmp : fieldDescription.split(",")) {
+      String[] fieldTuple = tmp.split(":(?=([^\"]*\"[^\"]*\")*[^\"]*$)", -1);
+      Field field = new Field();
+      field.setName(fieldTuple[0]);
+      field.setType(fieldTuple[1]);
+      headers.add(fieldTuple[0]);
+      fieldWidth.add(Integer.parseInt(fieldTuple[2]));
+      if (field.getType() == FIELD_TYPE.DATE) {
+        if (fieldTuple.length > 3) {
+          fieldDateFormat.put(field.getName(), fieldTuple[3].replace("\"", ""));
+        } else {
+          logger.error("Date format is missing for the field {}", field.getName());
+          throw new RuntimeException("Missing date format");
+        }
+      }
+      fields.add(field);
+    }
+    header = headers.toArray(new String[headers.size()]);
+    width = Ints.toArray(fieldWidth);
+    FixedWidthFieldLengths lengths = new FixedWidthFieldLengths(header, width);
+    FixedWidthParserSettings settings = new FixedWidthParserSettings(lengths);
+    settings.getFormat().setPadding(this.padding);
+    fixedWidthParser = new FixedWidthParser(settings);
+  }
+
+  private Object getValue(Field field, String value)
+  {
+    if (value == null) {
```

Comment:
Should this also check for empty string? Or do you treat empty string as a valid value?